node/libs/roles/src/validator/messages/consensus.rs

```diff
@@ -128,15 +161,37 @@ impl ValidatorSet {
         self.get(index).unwrap().clone()
     }
 
-    /// Signature threshold for this validator set.
+    /// Signature threshold for this validator committee.
     pub fn threshold(&self) -> usize {
-        threshold(self.len())
+        Self::THRESHOLD
     }
 
-    /// Maximal number of faulty replicas allowed in this validator set.
+    /// Maximal number of faulty replicas allowed in this validator committee.
     pub fn faulty_replicas(&self) -> usize {
         faulty_replicas(self.len())
     }
+
+    /// Compute the sum of signers weights.
+    pub fn weight_from_signers(&self, signers: Signers) -> usize {
+        self.weights
+            .iter()
+            .enumerate()
+            .filter(|(i, _)| signers.0[*i])
+            .map(|(_, weight)| weight)
+            .sum()
+    }
+
+    /// Compute the sum of signers weights.
+    pub fn weight_from_msgs<T: Variant<Msg>>(&self, signed: &[&validator::Signed<T>]) -> usize {
```

Comment:
this method doesn't take into account the possibility that the same signer occurs more than once.